File: Miscellaneous_Combination.java
package miscellaneous;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Miscellaneous {

    public static void main(String[] args) {
        List<String> goods = new ArrayList<>();
        goods.add("Eno");
        goods.add("Chidi");
        goods.add("Olu");
        goods.add("Ahmed");
        goods.add("Osas");
        goods.add("Gbeda");
        Combination combo = new Combination();
        List<String[]> result = combo.possibleWordCombinations(goods, 4);
        System.out.println(combo.words + " combination " + combo.r + ":\n");
        for (String[] set : result) {
            System.out.println(Arrays.toString(set) + ";");
        }
        System.out.println("\nNumber of ways is " + result.size() + ".");
    }

}

File: Miscellaneous_Combination.pl
#!/usr/bin/perl;

use strict;
use warnings;

use COMBINATION;

my @subjects = ("Eno", "Chidi", "Olu", "Ahmed", "Osas", "Gbeda");
my $r = 3;

# Use the combination module/class
my $combo = COMBINATION->new();

# $result receives an array reference of references
my $result = $combo->possibleWordCombinations(\@subjects, $r);

print ("[", join(", ", @subjects), "] combination ", $r, ":\n\n");
# for each array reference in a dereferenced '$result'
print (join(", ", @{$_}), "\n") for @{$result};
print ("\n\nNumber of ways is ", scalar @{$result}, ".");

print "\n\n";